api/controller/subscription.go

```diff
@@ -62,12 +63,41 @@ func CreateSubscription(dataBase moira.Database, auth *api.Authorization, userLo
 	}
 	subscription.TeamID = teamID
 	data := moira.SubscriptionData(*subscription)
+	if len(data.Tags) > 0 {
+		areTagsOneKind, err := areTagsAllSystemOrNot(data.Tags, dataBase)
+		if err != nil {
+			return api.ErrorInternalServer(err)
+		}
+		if !areTagsOneKind {
+			return api.ErrorInvalidRequest(fmt.Errorf("subscription tags should be only user-defined or only system"))
+		}
+	}
 	if err := dataBase.SaveSubscription(&data); err != nil {
 		return api.ErrorInternalServer(err)
 	}
 	return nil
 }
 
+func areTagsAllSystemOrNot(tags []string, dataBase moira.Database) (bool, error) {
```

Comment:
Мне кажется, по названию совсем неочевидно что функция делает, может попробовать как-то так?

```suggestion
func areAllTagsOfSameKind(tags []string, dataBase moira.Database) (bool, error) {
```

Reply:
Во, я долго голову ломал над названием :smile: